nucliadb_vectors2/src/vectors/data_point/ops_hnsw.rs

```diff
@@ -114,17 +112,25 @@ impl<'a, DR: DataRetriever> HnswOps<'a, DR> {
         self.tracker.consine_similarity(x, y)
     }
     fn closest_up_node<L: Layer>(
-        &self,
+        &'a self,
+        solution: &mut HashSet<Address>,
+        mut filters: impl Iterator<Item = &'a [u8]>,
```

Comment:
I think we could pass a slice here directly (since the `search` method takes a slice too as argument) and then avoid an useless copy at call site.